In [1]:
# Import the required libraries
import numpy as np  # linear algebra
import pandas as pd  # data processing
import plotly.express as px  # data visualization

In [2]:
# Define the list of columns to drop
drop_columns = ['Country/Region', 'Ship Country/Region', 'Percent change', 'Customer ID', 'Ship Address', 'Ship City', 'Ship State', 'State']

# Read the data and drop the irrelevant columns
sales_df = pd.read_csv('Sales_analytics.csv', usecols=lambda col: col not in drop_columns, encoding='latin-1')

In [3]:
# Covert Order Date and Shipped Date to datetime format
sales_df['Order Date'] = pd.to_datetime(sales_df['Order Date'], format='%d/%m/%Y')
sales_df['Shipped Date'] = pd.to_datetime(sales_df['Shipped Date'], format='%d/%m/%Y')

In [4]:
# Print the first 5 rows of the sales_df.
sales_df.head()

Order ID Order Date   Customer Name       City       Salesperson Region  \
0      1001 2019-01-27      CitrusFizz  Las Vegas  Mariya Sergienko   West   
1      1002 2019-01-27      CitrusFizz  Las Vegas  Mariya Sergienko   West   
2      1003 2019-01-04  Liquid Harmony   New York    Andrew Cencini   East   
3      1004 2019-01-04  Liquid Harmony   New York    Andrew Cencini   East   
4      1005 2019-01-04  Liquid Harmony   New York    Andrew Cencini   East   

  Shipped Date          Shipper Name      Ship Name Payment Method  \
0   2019-01-29    Shipping Company B      Karen Toh          Check   
1   2019-01-29    Shipping Company B      Karen Toh          Check   
2   2019-01-06  Shipping QuenchCraft  Christina Lee    Credit Card   
3   2019-01-06  Shipping QuenchCraft  Christina Lee    Credit Card   
4   2019-01-06  Shipping QuenchCraft  Christina Lee    Credit Card   

   Product Name            Category  Unit Price  Quantity   Sales  \
0          Beer           Beverages        14.0      49.0   686.0   
1   Dried Plums  Dried Fruit & Nuts         3.5      47.0   164.5   
2   Dried Pears  Dried Fruit & Nuts        30.0      69.0  2070.0   
3  Dried Apples  Dried Fruit & Nuts        53.0      89.0  4717.0   
4   Dried Plums  Dried Fruit & Nuts         3.5      11.0    38.5   

   Shipping Fee  
0       66.5420  
1       16.6145  
2      198.7200  
3      448.1150  
4        3.7345

In [5]:
# Print the last 5 rows of the sales_df
sales_df.tail(5)

Order ID Order Date   Customer Name         City       Salesperson  \
364      1428 2019-12-29     BubblyBurst       Denver         Jan Kotas   
365      1429 2019-12-06    SparkleCraft    Milwaukee   Michael Neipper   
366      1430 2019-12-06    SparkleCraft    Milwaukee   Michael Neipper   
367      1431 2019-12-04  Liquid Harmony     New York    Andrew Cencini   
368      1432 2019-12-03     BubblyBurst  Los Angelas  Mariya Sergienko   

    Region Shipped Date          Shipper Name               Ship Name  \
364   West   2019-12-31    Shipping Company B            Soo Jung Lee   
365  North   2019-12-08  Shipping BubblyBurst  Francisco Pérez-Olaeta   
366  North   2019-12-08  Shipping BubblyBurst  Francisco Pérez-Olaeta   
367   East   2019-12-08  Shipping BubblyBurst           Christina Lee   
368   West   2019-12-08  Shipping BubblyBurst            Thomas Axerr   

    Payment Method    Product Name            Category  Unit Price  Quantity  \
364          Check  Fruit Cocktail         Fruit & Veg       39.00      54.0   
365          Check     Dried Pears  Dried Fruit & Nuts       30.00      33.0   
366          Check    Dried Apples  Dried Fruit & Nuts       53.00      34.0   
367          Check         Gnocchi               Pasta       38.00      59.0   
368          Check       Green Tea           Beverages        2.99      24.0   

       Sales  Shipping Fee  
364  2106.00     214.81200  
365   990.00      95.04000  
366  1802.00     185.60600  
367  2242.00     226.44200  
368    71.76       7.10424

In [6]:
# Get the number of rows and columns
sales_df.shape

(369, 16)

In [7]:
# Check for missing values
sales_df.isnull().sum().sort_values(ascending=False)

Unit Price        3
Quantity          3
Order ID          0
Order Date        0
Customer Name     0
City              0
Salesperson       0
Region            0
Shipped Date      0
Shipper Name      0
Ship Name         0
Payment Method    0
Product Name      0
Category          0
Sales             0
Shipping Fee      0
dtype: int64

In [8]:
# Check the datatype
sales_df.dtypes

Order ID                   int64
Order Date        datetime64[ns]
Customer Name             object
City                      object
Salesperson               object
Region                    object
Shipped Date      datetime64[ns]
Shipper Name              object
Ship Name                 object
Payment Method            object
Product Name              object
Category                  object
Unit Price               float64
Quantity                 float64
Sales                    float64
Shipping Fee             float64
dtype: object

In [9]:
# Get the columns names
sales_df.columns

Index(['Order ID', 'Order Date', 'Customer Name', 'City', 'Salesperson',
       'Region', 'Shipped Date', 'Shipper Name', 'Ship Name', 'Payment Method',
       'Product Name', 'Category', 'Unit Price', 'Quantity', 'Sales',
       'Shipping Fee'],
      dtype='object')

In [10]:
# # Descriptive statistics for numerical variables
ds = ['Unit Price','Quantity','Sales','Shipping Fee']
sales_df[ds].describe().T.round(2)

count     mean      std    min     25%     50%      75%      max
Unit Price    366.0    21.43    16.89   2.99    9.65   14.00    34.80    81.00
Quantity      366.0    56.26    26.73  10.00   33.25   55.00    80.00   100.00
Sales         369.0  1178.96  1164.68   0.00  368.00  848.00  1458.00  7938.00
Shipping Fee  369.0   118.33   116.47   3.73   38.02   85.47   144.04   769.99

In [11]:
# Descriptive statistics for object variables
sales_df.describe(include='object').T

count unique                   top freq
Customer Name    369     14        Froth & Fusion   45
City             369     12               Chicago   57
Salesperson      369      8           Anne Larsen   87
Region           369      4                 North  117
Shipper Name     369      3  Shipping BubblyBurst  162
Ship Name        369     15         Roland Wacker   45
Payment Method   369      3           Credit Card  211
Product Name     369     24           Curry Sauce   34
Category         369     14             Beverages   91

In [12]:
# Number of transactions using 'Order ID'
num_transactions = sales_df['Order ID'].nunique()
print("Number of transactions:", num_transactions)

Number of transactions: 369


In [13]:
# Total Sales
Total_sales = sales_df['Sales'].sum().round()
print('Total Sales:',Total_sales)

Total Sales: 435036.0


### Summary of the key metrics:
+ Total sales is $435036.0
+ Total number of transactions is 369
+ Unit Price: The average price of items sold is $21.43, with a minimum of $2.99 and a maximum of $81.00.
+ Quantity: Each transaction involves selling an average of approximately 56.26 items, ranging from 10 to 100 items per transaction.
+ Sales: The average sales amount per transaction is $1178.96. Some transactions have no sales value ($0.00), while the highest recorded revenue from a single transaction is $7938.00.
+ Shipping Fee: The average shipping fee per transaction is $118.33, with fees varying from $3.73 to $769.99. 
+ There are 369 rows and 16 columns, and the categorical variables have varying numbers of unique categories, with some notable top occurrences such as "Froth & Fusion" for customer name, "Chicago" for city, "Anne Larsen" for salesperson, "North" for region, "Shipping BubblyBurst" for shipper name, "Roland Wacker" for ship name, "Credit Card" for payment method, "Curry Sauce" for product name, and "Beverages" for the category.

*These metrics provide valuable insights into the typical values and ranges for unit price, quantity, sales, and shipping fees in the dataset.*

# Data Visualization

In [14]:
sales_df.groupby('Customer Name')['Order ID'].size()

Customer Name
BlissBrew         32
BrewVibe          30
BubblyBurst       41
CitrusFizz         4
Company G          7
Company L          6
Froth & Fusion    45
InfuZest          26
Liquid Harmony    31
QuenchCraft       33
SparkleCraft      36
Splashology       12
Tasty Tingles     41
Zephyr Drinks     25
Name: Order ID, dtype: int64

In [15]:
# Count values of Customer Name
customer_count = sales_df['Customer Name'].value_counts().reset_index()
customer_count.rename(columns={'index':'Customer Name', 'Customer Name':'Count'}, inplace=True)

#  Bar chart of Customer Name
px.bar(customer_count,
       x='Customer Name',
       y='Count',
       color='Customer Name',
       title='Count of Transactions by Customers')

In [16]:
# Count values for City
city_count = sales_df['City'].value_counts().reset_index()
city_count.rename(columns={'index':'City', 'City':'Count'}, inplace=True)

#  Bar chart of City count
px.bar(city_count,
       x='City',
       y='Count',
       color='City',
       title='Number of Transactions Per Cities')



In [17]:
# Count values for Salesperson
salesperson_count = sales_df['Salesperson'].value_counts().reset_index()
salesperson_count.rename(columns={'index':'Salesperson', 'Salesperson':'Count'}, inplace=True)

#  Bar chart of Salesperson count
px.bar(salesperson_count,
       x='Salesperson',
       y='Count',
       color='Salesperson',
       title='Count of Transactions Per Salespersons')

In [18]:
# Count values for Region
region_count = sales_df['Region'].value_counts().reset_index()
region_count.rename(columns={'index':'Region', 'Region':'Count'}, inplace=True)

#  Pie chart of Region count
px.pie(region_count, 
       values='Count', 
       names='Region', 
       color='Region', 
       title='Transactions by Regions')
       

In [19]:
# Count values for Shipper Name
shipper_name_count = sales_df['Shipper Name'].value_counts().reset_index()
shipper_name_count.rename(columns={'index':'Shipper Name', 'Shipper Name':'Count'}, inplace=True)

#  Pie chart of Shipper Name count
px.pie(shipper_name_count,
       values='Count',
       names='Shipper Name',
       title='Transactions by Shippers')


In [20]:
# Count values for Ship Name
ship_name_count = sales_df['Ship Name'].value_counts().reset_index()
ship_name_count.rename(columns={'index':'Ship Name', 'Ship Name':'Count'}, inplace=True)

#  Bar chart of Ship Name count
px.bar(ship_name_count,
       x='Ship Name',
       y='Count',
       color='Ship Name',
       title='Count of Transactions by Ships')

In [21]:
# Count values for Payment Method
pay_method_count = sales_df['Payment Method'].value_counts().reset_index()
pay_method_count.rename(columns={'index':'Payment Method', 'Payment Method':'Count'}, inplace=True)

#  Pie chart of Payment Method count
px.pie(pay_method_count,
       values='Count',
       names='Payment Method',
       title= 'Transactions by Payment Methods')

In [22]:
# Count values for Product Name
product_name_count = sales_df['Product Name'].value_counts().reset_index()
product_name_count.rename(columns={'index':'Product Name', 'Product Name':'Count'}, inplace=True)

#  Bar chart of Product Name count
px.bar(product_name_count.head(),
       x='Product Name',
       y='Count',
       color='Product Name',
       title='Top 5 Most Common Products')

In [23]:
#  Bar chart of Product Name count
px.bar(product_name_count.tail().sort_values('Count'),
       x='Product Name',
       y='Count',
       color='Product Name',
       title='Bottom 5 Most Common Products')


In [24]:
# Count values for Category
category_count = sales_df['Category'].value_counts().reset_index()
category_count.rename(columns={'index':'Category', 'Category':'Count'}, inplace=True)

#  Bar chart of Category count
px.bar(category_count,
       x='Category',
       y='Count',
       color='Category',
       title='Most Common Product by Category')

In [25]:
# Extract Month_name from Order Date
sales_df['Month_name'] = pd.to_datetime(sales_df['Order Date']).dt.strftime('%b')

In [26]:
# Plot line chart of sales data with arranged months
px.bar(sales_df,
       x="Month_name",
       y="Sales",
       color="Month_name",
       title="Monthly Sales Trend")


In [27]:
# Group by Product Name and Sum the Sales
product_sales = sales_df.groupby('Product Name')['Sales'].sum().reset_index().sort_values(by='Sales', ascending=False)

#  Bar chart of Product Name
px.bar(product_sales.head(10),
       x='Product Name',
       y='Sales',
       color='Product Name',
       title='Top 10 Selling Products')

In [28]:
#  Bar chart of Product Name
px.bar(product_sales.tail(10).sort_values(by='Sales', ascending=True),
       x='Product Name',
       y='Sales',
       color='Product Name',
       title='Bottom 10 Selling Products')


In [29]:
# Group by Category and Sum the Sales
category_sales = sales_df.groupby('Category')['Sales'].sum().reset_index().sort_values(by='Sales', ascending=False)

#  Bar chart of Category
px.bar(category_sales,
       x='Category',
       y='Sales',
       color='Category',
       title='Best Selling Products by Category')

In [30]:
# Group by Salesperson and Sum the Sales
salesperson_sales = sales_df.groupby('Salesperson')['Sales'].sum().reset_index().sort_values(by='Sales', ascending=False)

#  Bar chart of Salesperson
px.bar(salesperson_sales,
       x='Salesperson',
       y='Sales',
       color='Salesperson',
       title='Total Sales by Salesperson')

In [31]:
# Group by City and Sum the Sales
city_sales = sales_df.groupby('City')['Sales'].sum().reset_index().sort_values(by='Sales', ascending=False)

#  Bar chart of City
px.bar(city_sales,
       x='City',
       y='Sales',
       color='City',
       title='Total Sales by City')

In [32]:
# Group by Region and Sum the Sales
region_sales = sales_df.groupby('Region')['Sales'].sum().reset_index().sort_values(by='Sales', ascending=False)

#  Bar chart of Region
px.bar(region_sales,
       x='Region',
       y='Sales',
       color='Region',
       title='Total Sales by Region')

In [33]:
# Group by Payment Method and Sum the Sales
pay_method_sales = sales_df.groupby('Payment Method')['Sales'].sum().reset_index().sort_values(by='Sales', ascending=False)

#  Bar chart of Payment Method
px.bar(pay_method_sales,
       x='Payment Method',
       y='Sales',
       color='Payment Method',
       title='Total Sales by Payment Method')

In [34]:
# Group by Customer and Sum the Sales
customer_sales = sales_df.groupby('Customer Name')['Sales'].sum().reset_index().sort_values(by='Sales', ascending=False)

# Bar chart of Customer
px.bar(customer_sales,
       x='Customer Name',
       y='Sales',
       color='Customer Name',
       title='Total Sales by Customer')

In [35]:
# Group by Shipper and Sum the Sales
shipper_sales = sales_df.groupby('Shipper Name')['Sales'].sum().reset_index().sort_values(by='Sales', ascending=False)

# Bar chart of Shipper
px.bar(shipper_sales,
       x='Shipper Name',
       y='Sales',
       color='Shipper Name',
       title='Total Sales by Shipper')

In [36]:
# Group by Shipper and Sum the Sales
ship_sales = sales_df.groupby('Ship Name')['Sales'].sum().reset_index().sort_values(by='Sales', ascending=False)

# Bar chart of Shipper
px.bar(ship_sales,
       x='Ship Name',
       y='Sales',
       color='Ship Name',
       title='Total Sales by Ship')

In [37]:
correlation_matrix = sales_df[['Unit Price', 'Quantity', 'Sales','Shipping Fee',]].corr().round(3)*100
print('Correlation Matrix:')
print(correlation_matrix)

Correlation Matrix:
              Unit Price  Quantity  Sales  Shipping Fee
Unit Price         100.0      -3.7   77.9          77.8
Quantity            -3.7     100.0   47.2          47.1
Sales               77.9      47.2  100.0          99.8
Shipping Fee        77.8      47.1   99.8         100.0
